Scripts/Runtime/Entities/TaskSystem/Util/ResolveTargetUtil.cs

```diff
@@ -0,0 +1,85 @@
+using Anvil.CSharp.Data;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Unity.Burst;
+using Unity.Collections;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    [BurstCompatible]
+    internal static class ResolveTargetUtil
+    {
+        // ReSharper disable once ClassNeverInstantiated.Local
+        private class ResolveTargetSharedStaticContext
+        {
+        }
+
+        [BurstCompatible]
+        // ReSharper disable once ClassNeverInstantiated.Local
+        private class ResolveTargetID<TResolveTargetType>
+            where TResolveTargetType : unmanaged, IEntityProxyInstance
+        {
+            // ReSharper disable once StaticMemberInGenericType
+            public static readonly SharedStatic<uint> ID = SharedStatic<uint>.GetOrCreate<ResolveTargetSharedStaticContext, TResolveTargetType>();
+        }
+
+        /// <summary>
+        /// The ID's for each ResolveTarget are not deterministic, it's a first come first serve basis.
+        /// As code changes through development, an older save file might have a different ID for the same type.
+        /// This mapping can be used to stitch them properly together. 
+        /// </summary>
+        /// TODO: Implement serialization
+        internal static readonly Dictionary<Type, uint> SERIALIZATION_MAPPING = new Dictionary<Type, uint>();
+
+        private static IDProvider s_IDProvider;
+
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.AfterAssembliesLoaded)]
+        private static void Init()
+        {
+            s_IDProvider?.Dispose();
+            s_IDProvider = new IDProvider();
+            SERIALIZATION_MAPPING.Clear();
+        }
+
+        public static uint RegisterResolveTarget<TResolveTargetType>()
+            where TResolveTargetType : unmanaged, IEntityProxyInstance
+        {
+            uint id = ResolveTargetID<TResolveTargetType>.ID.Data;
```

Comment:
Why would we have our own IDs for this? Wouldn't the entity being proxied, task type, and resolution type be enough?

Reply:
We must have ID's for runtime use because we can't use `Type` in Burst because the Type class is managed. However, using the Type as a generic does allow us to target a type specific class which we can convert to uint that is allowed to be used in Burst.

This allows us to write bursted client job code like so:

```csharp
instance.Resolve(new TimerCompletedInstance(instance.Entity),
                                 ref updater);
```

Which will take `TimerCompletedInstance` and convert to a uint which can then be used to lookup in an `UnsafeParallelHashMap` to get the right pointer for where to write the result to. 

In managed land we'd represent it via:

`Dictionary<Type, Action>`

But we have to represent it as:

`UnsafeParallelHashMap<uint, long>`

For Burst to work.

Reply:
oh, ok I thought this was for serialization to disk.